### ISSUE TITLE
Add JSON-LD from Culture Gaspésie to Artsdata

### ISSUE THREAD

**Issue by tammy-culture** [culturecreates/artsdata-orion#81]
Please grab JSON-LD from the following website (https://culturegaspesie.org/evenements/) and add it to Artsdata.ca.

**Comment by saumier** [comment]
@dev-aravind I added the workflow yml file, but it is not working. Please investigate.

**Comment by dev-aravind** [comment]
@saumier The event URLs are not available in the main HTML as they are doing an AJAX call to get the event information. How do you want to proceed on this?

**Comment by saumier** [comment]
@dev-aravind Please provide a rough estimate to enable the first step of crawling the event URLs using a headless browser (to enable Javascript). Thx.

**Comment by dev-aravind** [comment]
Estimate: 

Update workflow inputs to add a new parameter - 1h
Update the docker image to add headless crawling for entity URLs- 2.5h
Testing - 1h

Total - 4.5h

**Comment by saumier** [comment]
@dev-aravind Thanks for this estimate. Based on a quick search on my side, I found another site needing a headless crawling (needing Javascript to run) to get the list of entities. The site is https://www.adc659.ca/members/. So I would like to go ahead with this enhancement to the workflow.

There is also a need to be able to do a dry-run to test the ability of the action.  On other tools, like huginn, there is a "dry-run" feature that you can use to check the parameters. This would be a very helpful when adding a new site.  When the "dry-run" is selected, the action could do only some of the work in order to confirm the parameters. I'd like to know your thoughts, but the basic idea is to get the list of entities (tests the CSS class and pagination) and confirms that it can load JSON-LD from the first entity in the list. 

As a first iteration we can let the user look into the GitHub actions to see the results. As a second iteration there could be a reportCallbackUrl for json responses to be sent back to the client. We haven't used the reportCallbackUrl but this would also be passed to the Artsdata Databus as well so a client could follow the progress of the data loading into Artsdata.  I am adding a button in Artsdata Nebula to trigger a Github Action, and this button works but is currently lacking feedback. Once the reportCallbackUrl is added to the Action then Nebula will be able to show the progress of the data load.

Please share your ideas and I'll open a new issue so we can develop these enhancements to our Github Action.

To summarize:
- add parameter for headless crawling of entity URLs (deploy and use with Culture Gaspésie)
- add dry-run parameter with logs inside runner (deploy and test)
- add reportCallbackUrl parameter (deploy and test with Nebula)

**Comment by dev-aravind** [comment]
@saumier The [data](https://kg.artsdata.ca/en/query/show?title=Event%20entities%20in%20culturegaspesie-org&sparql=list_events&graph=http://kg.artsdata.ca/culture-creates/artsdata-orion/culturegaspesie-org) is up in artsdata using headless url crawl, but we can only fetch the events that load at the first page of the website as of now (we have to click the "see more events"  button in the browser to load more events). What do you propose?

**Comment by dev-aravind** [comment]
@saumier The place data is visible when I click the [JSON](https://kg.artsdata.ca/en/entity.jsonld?uri=https%3A%2F%2Fculturegaspesie.org%2Ftemporary%2F21c9b9d6-ce92-462a-9acc-6b63a9c84611) link. Can this be an issue with Nebula?

**Comment by dev-aravind** [comment]
Task for @dev-aravind : Find out why the Virtual Location URL is blocked, document if possible.

**Comment by dev-aravind** [comment]
@saumier The virtual-location issue is now solved.

**Comment by dev-aravind** [comment]
@saumier [This](https://github.com/culturecreates/artsdata-pipeline-action/commit/2ebfffb72d17e59b9864fd6dcbdba759e467a03c) is the commit.

**Comment by saumier** [comment]
@dev-aravind I added a comment to the SPARQL about why this error happens. Please read it and let me know by assigning this issue back to me. Thx.

**Comment by dev-aravind** [comment]
@saumier Noted. Thanks for the comment.
